### PR TITLE
Add PSC Usage and Controller metrics

### PR DESCRIPTION
**Pull request by swetharepakula**

### CODE REVIEW

File: pkg/metrics/types.go [hunk 1]
```diff
@@ -95,3 +101,8 @@ type L4ILBMetricsCollector interface {
 	// DeleteL4ILBService removes the given L4 ILB service key.
 	DeleteL4ILBService(svcKey string)
 }
+
+type PSCMetricsCollector interface {
```
Comment (freehan) [inline pkg/metrics/types.go:105]: put this psc pkg

File: pkg/psc/controller.go [hunk 2]
```diff
@@ -519,3 +555,7 @@ func validateUpdate(existingSA, desiredSA *alpha.ServiceAttachment) error {
 func SvcAttachmentKeyFunc(namespace, name string) string {
 	return fmt.Sprintf("%s/%s", namespace, name)
 }
+
+func ignoreConfigurationErrors(err error) error {
```
Comment (freehan) [inline pkg/psc/controller.go:559]: remove this?

**Review by freehan — COMMENTED** [review]
just nits. Otherwise LGTM

File: pkg/psc/controller.go (hunk 2 above)
Comment (swetharepakula) [reply pkg/psc/controller.go]: thanks! I missed removing this

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**